src/rust/engine/process_execution/src/lib.rs

```diff
@@ -210,7 +215,7 @@ pub struct ExecuteProcessRequest {
   /// see https://github.com/pantsbuild/pants/issues/6416.
   ///
   pub jdk_home: Option<PathBuf>,
-  pub target_platform: Platform,
+  pub target_platform: PlatformConstraint, //TODO I don't think we need this attr here
```

Comment:
The reason this is here is to distinguish cache entries keyed on the same request in the lmdb but which when run on different platforms would produce different output. The canonical example in pants being a pex build.